flower_shop/models/stock_lot.py

```diff
@@ -0,0 +1,34 @@
+from odoo import fields, models, api
+from datetime import datetime, timedelta
+from odoo.exceptions import ValidationError
+
+
+class StockLot(models.Model):
+    _inherit = 'stock.lot'
+
+    flower_water_id = fields.One2many('flower.water', 'lot_id')
+
+    def water_flower(self):
+        self.write({'flower_water_id': [(0, 0, {'lot_id': self.id, 'watering_datetime': datetime.now()})]})
+
+
+    @api.constrains('flower_water_id', 'flower_water_id.watering_datetime')
+    def prevent_watering_before_time(self):

```

Comment:
naming convention is usually **def \_constrain\_{CONCATENATED_FIELD_NAMES}**, example: **def _constrain_flower_watering_datetime**